src/main/java/slimeknights/tconstruct/fluids/TinkerFluids.java

```diff
@@ -1,53 +1,32 @@
-  private static final DeferredRegister<Fluid> FLUIDS = new DeferredRegister<>(ForgeRegistries.FLUIDS, TConstruct.modID);
+  private static final FluidDeferredRegister FLUIDS = new FluidDeferredRegister(TConstruct.modID);
 
-  public static RegistryObject<FlowingFluid> blue_slime_fluid = FLUIDS.register("blue_slime_fluid", () -> new SlimeFluid.Source(TinkerFluids.blue_slime_fluid_properties));
-  public static RegistryObject<FlowingFluid> blue_slime_fluid_flowing = FLUIDS.register("blue_slime_fluid_flowing", () -> new SlimeFluid.Flowing(TinkerFluids.blue_slime_fluid_properties));
-  public static RegistryObject<FlowingFluidBlock> blue_slime_fluid_block = BLOCKS.register("blue_slime_fluid_block", () -> new FlowingFluidBlock(blue_slime_fluid, Block.Properties.create(Material.WATER).doesNotBlockMovement().hardnessAndResistance(100.0F).noDrops()));
-  public static RegistryObject<Item> blue_slime_fluid_bucket = ITEMS.register("blue_slime_bucket", () -> new BucketItem(blue_slime_fluid, new Item.Properties().containerItem(Items.BUCKET).maxStackSize(1).group(ItemGroup.MISC)));
-  public static final SlimeFluid.Properties blue_slime_fluid_properties = new SlimeFluid.Properties(blue_slime_fluid, blue_slime_fluid_flowing, FluidAttributes.builder(FluidIcons.FLUID_STILL, FluidIcons.FLUID_FLOWING).color(0xef67f0f5).density(1500).viscosity(1500).temperature(310)).explosionResistance(100.0F).bucket(blue_slime_fluid_bucket).block(blue_slime_fluid_block);
-
-  public static RegistryObject<FlowingFluid> purple_slime_fluid = FLUIDS.register("purple_slime_fluid", () -> new SlimeFluid.Source(TinkerFluids.purple_slime_fluid_properties));
-  public static RegistryObject<FlowingFluid> purple_slime_fluid_flowing = FLUIDS.register("purple_slime_fluid_flowing", () -> new SlimeFluid.Flowing(TinkerFluids.purple_slime_fluid_properties));
-  public static RegistryObject<FlowingFluidBlock> purple_slime_fluid_block = BLOCKS.register("purple_slime_fluid_block", () -> new FlowingFluidBlock(purple_slime_fluid, Block.Properties.create(Material.WATER).doesNotBlockMovement().hardnessAndResistance(100.0F).noDrops()));
-  public static RegistryObject<Item> purple_slime_fluid_bucket = ITEMS.register("purple_slime_bucket", () -> new BucketItem(purple_slime_fluid, new Item.Properties().containerItem(Items.BUCKET).maxStackSize(1).group(ItemGroup.MISC)));
-  public static final SlimeFluid.Properties purple_slime_fluid_properties = new SlimeFluid.Properties(purple_slime_fluid, purple_slime_fluid_flowing, FluidAttributes.builder(FluidIcons.FLUID_STILL, FluidIcons.FLUID_FLOWING).color(0xefd236ff).density(1600).viscosity(1600).temperature(370)).explosionResistance(100.0F).bucket(purple_slime_fluid_bucket).block(purple_slime_fluid_block);
+  public static final FluidObject<ForgeFlowingFluid> blue_slime_fluid = FLUIDS.register("blue_slime", FluidAttributes.builder(FluidIcons.FLUID_STILL, FluidIcons.FLUID_FLOWING).color(0xef67f0f5).density(1500).viscosity(1500).temperature(310), Material.WATER);
```

Comment:
Slime fluid should remain, use `SlimeFluid.Still::new` and `SlimeFluid.Flowing::new` as parameters after the attributes.

Reply:
Yeah. I wasn't sure how to make use of your fluid reg at the moment I wrote this. It
*does* work tho.

Reply:
The base Forge fluid "works", but if you check the slime fluid class, it has some extra logic in there (I think related to slime spawning)